### PR TITLE
replace train_percent_check with limit_train_batches

### PR DESCRIPTION
**Pull request by williamFalcon**
Fixes #2213

### CODE REVIEW

File: pytorch_lightning/trainer/trainer.py [hunk 1]
```diff
@@ -102,10 +102,10 @@ def __init__(
         min_epochs: int = 1,
         max_steps: Optional[int] = None,
         min_steps: Optional[int] = None,
-        train_percent_check: float = 1.0,
+        limit_train_batches: Union[int, float] = 1.0,
```
Comment (Borda) [inline pytorch_lightning/trainer/trainer.py:105]: back compatibility

Comment (williamFalcon) [reply pytorch_lightning/trainer/trainer.py:105]: ? this was fixed

File: pytorch_lightning/trainer/trainer.py [hunk 2]
```diff
@@ -494,7 +494,9 @@ def __init__(
 
         # how much of the data to use
         # TODO: remove in 0.10.0
-        if overfit_pct > 0:
+        if overfit_pct is not None:
+            rank_zero_warn("Argument `overfit_pct` is now set by `overfit_batches` since v0.8.0"
+                           " and this argument will be removed in v0.10.0", DeprecationWarning)
```
Comment (williamFalcon) [inline pytorch_lightning/trainer/trainer.py:499]: 1.0.0
```suggestion
                           " and this argument will be removed in v1.0.0", DeprecationWarning)
```

Comment (Borda) [reply pytorch_lightning/trainer/trainer.py:499]: I would rather keep smaller granularity and if finally goes with 1.0 right after 0.9 we id very trivial to do all these cleaning compare to having everything as 1.0 and no ide when it was really meant to replace...

File: pytorch_lightning/callbacks/gradient_accumulation_scheduler.py [hunk 3]
```diff
@@ -1,5 +1,5 @@
 r"""
-Gradient Accumulator
+GradientAccumulator
```
Comment (Borda) [inline pytorch_lightning/callbacks/gradient_accumulation_scheduler.py:2]: why together? it is not even linked to a class, isn't it?

Comment (williamFalcon) [reply pytorch_lightning/callbacks/gradient_accumulation_scheduler.py:2]: yeah, it should be separate

Comment (Borda) [reply pytorch_lightning/callbacks/gradient_accumulation_scheduler.py:2]: ok, there are a few more...

File: pytorch_lightning/callbacks/early_stopping.py [hunk 4]
```diff
@@ -1,6 +1,6 @@
 r"""
-Early Stopping
-==============
+EarlyStopping
+=============
```
Comment (Borda) [inline pytorch_lightning/callbacks/early_stopping.py:3]:
```suggestion
Early Stopping
==============
```

File: pytorch_lightning/callbacks/lr_logger.py [hunk 5]
```diff
@@ -1,7 +1,7 @@
 r"""
 
-Logging of learning rates
-=========================
+LearningRateLogger
+==================
```
Comment (Borda) [inline pytorch_lightning/callbacks/lr_logger.py:4]:
```suggestion
Learning Rate Logger
====================
```

File: pytorch_lightning/trainer/trainer.py [hunk 6]
```diff
@@ -486,27 +493,37 @@ def __init__(
...
+        # TODO: remove in 0.10.0
+        if train_percent_check is not None:
+            rank_zero_warn("Argument `train_percent_check` is now set by `limit_train_batches` since v0.8.0"
+                           " and this argument will be removed in v0.10.0", DeprecationWarning)
+            limit_train_batches = train_percent_check
 
-        # TODO: convert train_percent_check to limit_train_batches
-        self.determine_data_use_amount(train_percent_check, limit_val_batches,
-                                       limit_test_batches, overfit_batches)
+        self.limit_test_batches = _determine_limit_batches(limit_test_batches)
```
Comment (williamFalcon) [inline pytorch_lightning/trainer/trainer.py:523]: this is unecessary and hides a very simple op. Please revert

Comment (Borda) [reply pytorch_lightning/trainer/trainer.py:523]: it prevents typo as it is used several time in the code

**Pull request merged**